src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasEnabledEntityValidator.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/*
+ * @author Gustavo Perdomo <gperdomor@gmail.com>
+ */
+class HasEnabledEntityValidator extends ConstraintValidator
+{
+    /**
+     * @var ManagerRegistry
+     */
+    protected $registry;
+    /**
+     * @var \Symfony\Component\PropertyAccess\PropertyAccessor
+     */
+    protected $accessor;
+
+    /**
+     * @param ManagerRegistry $registry
+     */
+    public function __construct(ManagerRegistry $registry)
+    {
+        $this->registry = $registry;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * @param object $entity
+     * @param Constraint $constraint
+     *
+     * @throws UnexpectedTypeException
+     * @throws ConstraintDefinitionException
+     */
+    public function validate($entity, Constraint $constraint)
+    {
+        if (!$constraint instanceof HasEnabledEntity) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__ . '\HasEnabledEntity');
+        }
+
+        $enabled = $this->accessor->getValue($entity, $constraint->enabledPath);
+
+        if ($enabled === true) {
+            return;
+        }
+
+        $entityManager = $this->getProperEntityManager($entity, $constraint);
+
+        $this->ensureEntityHasProvidedEnabledField($entity, $constraint, $entityManager);
+
+        $criteria = array($constraint->enabledPath => true);
+
+        $repository = $entityManager->getRepository(get_class($entity));
+        $result = $repository->{$constraint->repositoryMethod}($criteria);
+
+        /* If the result is a MongoCursor, it must be advanced to the first
+         * element. Rewinding should have no ill effect if $result is another
+         * iterator implementation.
+         */
+        if ($result instanceof \Iterator) {
+            $result->rewind();
```

Comment:
Are you sure? because i created a iterator object and move some position and when call the reset method as you said, the return was the current index/position of the iterator (not the value in that position)